File: src/problems/leetcode-cn/1653. 使字符串平衡的最少删除次数/solution/java/Solution.java
class Solution {
    public int minimumDeletions(String s) {
        int length = s.length();
        if (length < 2) return 0;

        int bc = 0, del = 0;

        for (byte b : s.getBytes()) {
            if (b == 'b') bc++;
            else
                if (++del > bc) del = bc;
        }

        return del;
    }
}

File: src/problems/leetcode-cn/1653. 使字符串平衡的最少删除次数/solution/rust/solution.rs
impl Solution {
    pub fn minimum_deletions(s: String) -> i32 {
        let length = s.len();
        if length < 2 {
            return 0;
        }

        let (mut bc, mut del) = (0, 0);

        for v in s.bytes() {
            if v == b'b' {
                bc += 1;
            } else {
                del += 1;
                if del > bc {
                    del = bc
                }
            }
        }

        del
    }
}